test/dotnet-cache.Tests/GivenDotnetCachesAndPublishesProjects.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.PlatformAbstractions;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.Publish.Tests
+{
+    public class GivenDotnetPublishPublisheswithCacheProjects : TestBase
+    {
+        static string tfm = "netcoreapp2.0";
+        static string frameworkVersion = new Muxer().SharedFxVersion;
+        [Fact]
+        public void ItPublishesARunnablePortableApp()
+        {
+            var testAppName = "SimpleDependencies";
+            var profileProjectName = "NewtonsoftFilterProfile";
+
+            var testInstance = TestAssets.Get(testAppName)
+                .CreateInstance()
+                .WithSourceFiles()
+                .UseCurrentRuntimeFrameworkVersion();
+
+            var testProjectDirectory = testInstance.Root.ToString();
+            var rid = DotnetLegacyRuntimeIdentifiers.InferLegacyRestoreRuntimeIdentifier();
+            var localAssemblyCache = Path.Combine(testProjectDirectory, "localAssemblyCache");
+            var workingDirectory = Path.Combine(testProjectDirectory, "workingDirectory");
+            var profileProject = Path.Combine(TestAssetsManager.AssetsRoot, profileProjectName, $"{profileProjectName}.csproj");
+
+            new RestoreCommand()
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should().Pass();
+
+            new CacheCommand()
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute($"--entries {profileProject} --framework {tfm} --runtime {rid} --output {localAssemblyCache} --working-dir {workingDirectory} --preserve-working-dir --framework-version {frameworkVersion}")
+                .Should().Pass();
+
+            var configuration = Environment.GetEnvironmentVariable("CONFIGURATION") ?? "Debug";
+           
+            new PublishCommand()
+                .WithFramework(tfm)
+                .WithWorkingDirectory(testProjectDirectory)
+                .WithProFileProject(profileProject)
+                .Execute()
+                .Should().Pass();
+
+            var outputDll = Path.Combine(testProjectDirectory, "bin", configuration, tfm, "publish", $"{testAppName}.dll");
+
+            new TestCommand("dotnet")
+                .WithEnvironmentVariable("DOTNET_SHARED_PACKAGES", localAssemblyCache)
```

Comment:
So, in order to use this feature, an environment variable will have to be set? Should we have, maybe, a parameter to the host that would allow us to pass the shared packages folder?